### PR TITLE
Fixed issue #18823: STATA Export fails with certain question types

### PR DESCRIPTION
**Pull request by Oz-men**
Fixed issue #18823: Stata export failed with question type "radio"

### CODE REVIEW

File: application/core/plugins/ExportSTATAxml/STATAxmlWriter.php [hunk 1]
```diff
@@ -401,7 +401,7 @@ protected function updateCustomresponsemap()
                             // For radio lists, user wants code, not label
                             // TODO: We could skip this loop if we had answer code
                             foreach ($this->customFieldmap['answers'][$iQID][$iScaleID] as $answer) {
-                                if ($answer['answer'] == $response) {
+                                if (is_array($answer) && isset($answer['answer']) && $answer['answer'] == $response) {
```
Comment (Shnoulle) [inline application/core/plugins/ExportSTATAxml/STATAxmlWriter.php:404]: `if (isset($answer['answer']) && $answer['answer'] == $response) {`

Check if it's an array not needed, just check if it exist.

**Review by Shnoulle — CHANGES_REQUESTED** [review]
Check if it's an array not needed

**Comment by Shnoulle** [comment]
PS: thank you :)

File: application/core/plugins/ExportSTATAxml/STATAxmlWriter.php (hunk 1 above)
Comment (Oz-men) [reply application/core/plugins/ExportSTATAxml/STATAxmlWriter.php]: Thank you! As requested I removed the array check.

**Review by Shnoulle — APPROVED** [review]
:+1:

**Comment by gabrieljenik** [comment]
I have tested this. A dump is not produced :)
Still, I can't evaluate the results of the file.
By the type of change here, I guess it is OK to merge?

Thgoughts @Shnoulle ? @olleharstedt ?

**Comment by Shnoulle** [comment]
> Still, not sure if this is relevant: https://bugs.limesurvey.org/view.php?id=18823#c75167

I think you're right about workaround, but search original issue take more time a lot more time …

Someone report another issue for a lack of answers or something like this in STATA export, then maybe we can fix it.

Here : reporter have issue and create a PR : i really think we must merge !

**Pull request merged**